ste/sender-blockBlob.go

```diff
@@ -176,7 +176,16 @@ func (s *blockBlobSenderBase) Cleanup() {
 		//    situation. Deletion has very different semantics then, compared to not deleting.
 		deletionContext, cancelFn := context.WithTimeout(context.Background(), 30*time.Second)
 		defer cancelFn()
-		_, _ = s.destBlockBlobURL.Delete(deletionContext, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
+		if jptm.WasCanceled() {
+			_, err := s.destBlockBlobURL.GetProperties(deletionContext, azblob.BlobAccessConditions{})
+
+			if stgErr, ok := err.(azblob.StorageError); ok && stgErr.ServiceCode() == azblob.ServiceCodeBlobNotFound {
+				_, _ = s.destBlockBlobURL.Delete(deletionContext, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
```

Comment:
Maybe pop a comment here to say that Delete can delete uncommited blobs (and that is in fact why we are doing it here).  Otherwise, its very counter intuitve for a failed existence test to be followed by an action on the "non-existent" object.

Reply:
Done